### PR TITLE
[UI] Add new icons

### PR DESCRIPTION
**Pull request by lachlanjc**
## Summary of the problem

I've designed a bunch of new icons I've added to the codebase!

## Describe your changes

On the nav, a few new icons:

- Updated the arrows on the Financials & Transfers to be more elegant
- New perks icon that's more abstract since we don't give you merch

- New team icon that communicates multiple people instead of one special person
- New settings icon that's distinct from the user settings page
- New reimbursements icon (vaguely in the shape of an R) that communicates the flow of money back & forth
- Since receipts are physically organized with paperclips, and they're also generic "attachments," using the paperclip icon for attachments
- Using the "money document" icon for invoices, since they are documents and no briefcases are involved

image

### CODE REVIEW

File: app/views/events/_nav.html.erb
Comment (sampoder) [inline app/views/events/_nav.html.erb]: Let's keep Transactions imo, it's more direct. I know that page's header is Financials but that's going to change to "Transactions" after we replace the home page.

File: app/views/events/_nav.html.erb
Comment (sampoder) [inline app/views/events/_nav.html.erb]: Looks sweet!

File: app/views/events/_nav.html.erb
Comment (sampoder) [inline app/views/events/_nav.html.erb]: Nice change

File: app/views/events/_nav.html.erb
Comment (sampoder) [inline app/views/events/_nav.html.erb]: We've chatted about this before in https://github.com/hackclub/hcb/issues/7324, the consensus seemed to be keeping briefcase even if it wasn't ideal with the goal of finding something better

**Review by sampoder — COMMENTED** [review]
Sorry looks like these comments never posted!

File: app/views/events/_nav.html.erb
Comment (lachlanjc) [reply app/views/events/_nav.html.erb]: I'd love to have a more exact icon, but I've struggled to design one that doesn't resemble Documents or Receipts. The verb with an invoice is to send—you send the invoice as a request, the recipient sends money back—so I think it's a far more appropriate choice than my previous selection of the briefcase icon, which doesn't correspond to any part of invoices. I've clarified the tooltip to make sure folks understand what these are.

Comment (sampoder) [reply app/views/events/_nav.html.erb]: +1 on struggling to think of a more exact icon. There are a fair few places where you can send something on HCB though, for example, sending an invite to join a team. I think we should avoid that ambiguity, at least the briefcase icon was unique to invoices. That made it work for places like the activity log or the command bar.

If we want to continue this discussion, is it cool if we pull it into a separate PR that way it doesn't block on getting everything else merged?

**Review by garyhtou — COMMENTED** [review]
+1 on @sampoder's feedback. Feel free to merge after the feedback is addressed!

Thanks for reviewing @sampoder; we're aligned here.

**Comment by garyhtou** [comment]
@lachlanjc I love all these new icons you have. They're going to come in very handy!

File: app/views/events/_nav.html.erb
Comment (lachlanjc) [reply app/views/events/_nav.html.erb]: I worked on a wide range of icons for invoices, most of them I didn't love. Here's the set, in various states of completion:
CleanShot 2024-08-31 at 13 54 10@2x

Adding wind to the send icon I think clarifies that you're doing the sending, so I'm pretty happy with this new set!
CleanShot 2024-08-31 at 13 53 08@2x

Comment (sampoder) [reply app/views/events/_nav.html.erb]: Screenshot 2024-08-31 at 1 25 36 PM

What is we used this icon but with a dollar sign on the document?

Comment (lachlanjc) [reply app/views/events/_nav.html.erb]: Unfortunately it's too visually busy at small sizes. I've dropped the envelope part, updated screenshot above!

File: app/views/events/_nav.html.erb
Comment (sampoder) [reply app/views/events/_nav.html.erb]: Hey, uhh, I was working on getting this merged into production but this is one of a couple things blocking, can you please reapply my commit?

File: app/views/events/_nav.html.erb
Comment (sampoder) [reply app/views/events/_nav.html.erb]: This is just the old receipts icon isn't it? I'm on the fence, @YodaLightsabr @malted what do you all think? My hesitation is that we're switching around things we've been using for a while but I do think the changes make sense.

Comment (lachlanjc) [reply app/views/events/_nav.html.erb]: Yep, it's not ideal to be shuffling icons around, but I do think the new set is a lot clearer overall. I'm not over the moon with this invoices icon, but it does communicate "money document", and luckily there's a clear legend of which icons are which with the labeled sidebar items. My guess is after a day with the new set it'll be pretty intuitive for customers.

File: app/controllers/events_controller.rb
Comment (garyhtou) [inline app/controllers/events_controller.rb:1]: Just as an FYI, I found that these `icon` keys are not referenced anywhere. I've created a ticket to remove them: https://github.com/hackclub/hcb/issues/8069

For the time being, we can keep your changes here; your call.

File: app/javascript/components/command_bar/actions.js
Comment (garyhtou) [inline app/javascript/components/command_bar/actions.js]: I like that we're updating the capitalization here.

I will mention that across the codebase, we use the term "Check deposit(s)" (as a noun). This would be the first instance where we're calling it "Deposit checks".

For consistency, I would like for us to keep this as "Check deposits". This change can happen in this PR or as a follow-up PR.

File: app/views/hcb_codes/_icon.html.erb
Comment (garyhtou) [inline app/views/hcb_codes/_icon.html.erb:1]: Can we please update the icon for Reimbursement on line 28 from `attachment` to `reimbursement`?

https://github.com/hackclub/bank/blob/5556509eb9ac642b76b03fa58f3ccc31b0516763/app/views/hcb_codes/_icon.html.erb#L28

File: app/views/events/_nav.html.erb
Comment (garyhtou) [inline app/views/events/_nav.html.erb]: With the current content on the Transactions page, I can see why we'd want to rename this page to "Financials".

**I want to take a step back and provide some context on how this PR fits into our roadmap.**

In the review comment above, I mentioned how we'd like to ship this PR with our **UI3** launch. This **UI3** launch includes a new [**Home** page](https://github.com/hackclub/hcb/pull/7105) for organizations. Here's a sneak peek.

| | Home | Transactions |
|-|-|-|
| **Before** | nonexistent (was Transactions 👉) |  |
| **After** (https://github.com/hackclub/hcb/pull/7105) |  The new **Home** page provides glanceable, widget-based information on your organization.<br/><br/>Designed to be suitable for both Transparency Mode and organizers, it offers a one-stop shop for understanding the current state of an organization's finances. Chips have been added to provide quick access for frequent actions (the final version will draw from https://github.com/hackclub/hcb/pull/7839). |  The **Transactions** page has been slimed down to provide you no-frills access to an organization's ledger.<br/><br/>Search and filter to your heart's desire. The balance graph and team members have been moved to the **Home** page. <br/><br/>**This page now only contains an index of transactions.** |

<br/>

With that in mind, given that this PR will be shipped alongside the new **Home** page, I believe it's best to keep the Transactions page named "Transactions".

File: app/views/events/_nav.html.erb
Comment (garyhtou) [inline app/views/events/_nav.html.erb]: I want to mention that with this icon change, this page will be sharing the same `bank-account` icon as the **Account numbers** page.

I think that's not a big detail, however, I do want to make sure that we're aware of this similarity. I'm happy for us to use the `bank-account` icon for this page.

There is additional discussion about removing the **Account numbers** page entirely. That may help solve the duplicate icon.

File: app/assets/images/icons/plus.svg
Comment (garyhtou) [inline app/assets/images/icons/plus.svg:1]: This is a very much-needed icon! Thank you!

This PR renames the previous image `plus` icon to image `plus-box`.

I like how you did not update the existing usages to use `plus-box`. `plus-box` was a bit heavy for our typical use cases (buttons for adding/creating something). The (new) `plus` does a better job of communicating the same purpose without the added weight/busyness.

However, I did find one location where I prefer `plus-box` over `plus`. We have a banner that provides instructions for installing the PWA app. In the screenshots below, let's take a look at the plus in the red banner.

| Before | After|
|-|-|
|  |  |

The goal of the `+` plus is to replicate Apple's icon for "Add to Home Screen". Looking at the screenshot below, we can see Apple's icon is most similar to `plus-box`.

image

I would recommend that we switch this usage from `plus` to `plus-box`.
https://github.com/hackclub/bank/blob/89a7d64e4367f83265efb8b16dba5e3a4b7ff6c2/app/views/users/_nav.html.erb#L58

Other than that, this is great!

**Review by garyhtou — COMMENTED** [review]
I've said this before and I'll say it again, I ABSOLUTELY LOVE all the new icons you have! They're super handy 🫶

Overall, there are some amazing changes here. I especially love the icons you picked for perks and teams. Those do a much better job of representing their respective pages. Also, thanks for the detailed PR description explaining the motivation behind each change.

I've left some very minor feedback. If you'd like to take a minute or two to address them, I'd appreciate it. As always, I'm happy to hear your thoughts on any feedback I left.

Also, I know you're in school and are very busy. So, if you'd prefer, I'd be happy to handle the feedback on your behalf. Let me know what would work best for you.

---

As for getting this shipped, Dylan may have already mentioned to you that we plan to include this in a large **UI3** launch. This PR, along with a couple of others, will be announced in a changelog post and shipped at the same time. The goal of **UI3** is to provide users a visual refresh and to make "breaking changes" to core concepts. (It also has the nice benefit of generating excitement and demonstrating forward progress to our users.)

Since this PR repurposes heavily used icons, we thought it was very fitting for the **UI3** launch.

| | Before | After|
|-|-|-|
|Reimbursement| image | image |
|Receipt| image | image |
|Invoice| image | image |

_The current Reimbursement icon becomes the Receipt icon. And the current Receipt icon becomes the Invoice icon._

With the large **UI3** launch/announcement, users will be mentally prepared for a new look. This will hopefully minimize any confusion from us switching the icons around!

We're headed in a great direction and I appreciate the work you're doing. As always, I'd be happy to hear any thoughts you have.

File: app/assets/images/icons/plus.svg
Comment (lachlanjc) [reply app/assets/images/icons/plus.svg:1]: Great catch, reverted!

File: app/views/events/_nav.html.erb
Comment (lachlanjc) [reply app/views/events/_nav.html.erb]: Yep, that's fine; #7839 redesigns the account numbers section & removes that page!

**Comment by garyhtou** [comment]
Hi @lachlanjc! I just wanted to check in to see if I may make the above-mentioned edits to this branch. I'd love to get this merged — same goes for [#7741](https://github.com/hackclub/hcb/pull/7741#issuecomment-2427655001)!

File: app/javascript/components/command_bar/actions.js
Comment (lachlanjc) [reply app/javascript/components/command_bar/actions.js]:
```suggestion
    name: 'Check deposits',
```

**Comment by lachlanjc** [comment]
I think we're good to go with this now @garyhtou! Sorry for the delay, HCB has been at the bottom of my to-do list during a very busy midterms season.

**Comment by garyhtou** [comment]
Thanks @lachlanjc! I totally get it— midterms are important! I'll do a final review pass on this and get it merged into our UI3 branch. From there, we'll test it with the rest of the UI3 PRs and prepare the changelog post.

Feel free to ping me once [#7741](https://github.com/hackclub/hcb/pull/7741#issuecomment-2427655001) is ready as well!

**Review by garyhtou — APPROVED** [review]
Great! Thank you. I'm now merging this into the UI3 branch

**Pull request merged**
